Example css fitting inputs

jake-abma opened the pull request:
New technique for Using Fitting Labels and Inputs for Reflow, you can preview [Using CSS max-width and height to fit images](https://rawgit.com/w3c/wcag/tech-fitting-labels-inputs-viewport/techniques/css/tech-fitting-labels-inputs-reflow.html), and there is an associated [example page](https://rawgit.com/w3c/wcag/example-css-fitting-inputs/working-examples/css-fitting-inputs/index.html).

alastc commented:
Hi Jake,

At first glance I think it is good, a couple of text-simplifications come to mind, that's all:

- Are you sure that it will work as a horizontally scrolling bit of content? If not, it would be fine to focus this technique on vertically scrolling content, so remove the ", or a vertical scroll bar at a height equivalent to 256 CSS pixels for content intended to scroll horizontally".
- The second paragraph seems a bit complicated, are you esentially saying: "Layouts can reflow as needed to display the region on the screen, so the available width for labels and inputs varies. The same form elements and CSS can be used to adapt to varying widths as shown in the example below."

From the call:
- Update the title to Flex rather than flexbox.
- Add link to the flex module reference in desc, and a flexbox resource in the resources section (e.g. MDN or something else instructional.)

Question from me: The media queries don't account for the width of the container, just the viewport. Is there method of querying by parent-element dimensions?

allanj-uaag commented:
This is really nice.
minor niggle. in Chrome (and perhaps safari) the select cuts off the bottom
of the option (banana). in the css

      label,
      select {
        height: 44px;
        font-size: 1rem;
        padding: 12px 0 14px 0;
      }

possible fixes:
remove the font-size, then it matches the other inputs. or
change height to min-height. fixes the problem in chrome, and has no ill
effects on IE, Edge, or FF

      label,
      select {
        *min-height:* 44px;
        font-size: 1rem;
        padding: 12px 0 14px 0;
      }


On Tue, Oct 16, 2018 at 11:43 AM Alastair Campbell <notifications@github.com>
wrote:

> Hi Jake,
>
> At first glance I think it is good, a couple of text-simplifications come
> to mind, that's all:
>
>    - Are you sure that it will work as a horizontally scrolling bit of
>    content? If not, it would be fine to focus this technique on vertically
>    scrolling content, so remove the ", or a vertical scroll bar at a height
>    equivalent to 256 CSS pixels for content intended to scroll horizontally".
>    - The second paragraph seems a bit complicated, are you esentially
>    saying: "Layouts can reflow as needed to display the region on the screen,
>    so the available width for labels and inputs varies. The same form elements
>    and CSS can be used to adapt to varying widths as shown in the example
>    below."
>
> From the call:
>
>    - Update the title to Flex rather than flexbox.
>    - Add link to the flex module reference in desc, and a flexbox
>    resource in the resources section (e.g. MDN or something else
>    instructional.)
>
> Question from me: The media queries don't account for the width of the
> container, just the viewport. Is there method of querying by parent-element
> dimensions?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/w3c/wcag/pull/510#issuecomment-430310200>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AG_WL98-YP54TLIbcrmiO4w--sKVCmvyks5ulgzEgaJpZM4XSyc3>
> .
>


-- 
Jim Allan, Accessibility Coordinator
Texas School for the Blind and Visually Impaired
1100 W. 45th St., Austin, Texas 78756
voice 512.206.9315    fax: 512.206.9452 http://www.tsbvi.edu/
"We shape our tools and thereafter our tools shape us." McLuhan, 1964

rachaelbradley commented:
Github was having issues during this comment and I am not sure how this was closed. Reopening.

jake-abma commented:
- paragraph starting "Layouts..." changed
- Removed: "Medium complex"

jake-abma commented:
Hi @alastc 

> Are you sure that it will work as a horizontally scrolling bit of content? If not, it would be fine to focus this technique on vertically scrolling content, so remove the ", or a vertical scroll bar at a height equivalent to 256 CSS pixels for content intended to scroll horizontally".

No, not sure, but also not sure the other way around. Checked several CJK  language sites but could't find examples. May be better to ask some people from i18n??? Deleted text though for now though!

> The second paragraph seems a bit complicated, are you esentially saying: "Layouts can reflow as needed to display the region on the screen, so the available width for labels and inputs varies. The same form elements and CSS can be used to adapt to varying widths as shown in the example below."

Changed

> From the call:
> 
> Update the title to Flex rather than flexbox. 

Not sure if this is better. The term "Flexbox"is much more well known and also used in the spec. Can change but what are the exact reasons to not use flex box?

> Add link to the flex module reference in desc, and a flexbox resource in the resources section (e.g. MDN or something else instructional.)

Added

> Question from me: The media queries don't account for the width of the container, just the viewport. Is there method of querying by parent-element dimensions?

Not sure exactly what you mean...?

jake-abma commented:
Text results updated, vertical removed

alastc commented:
As this is the example for the associated technique (#509), I've merged this into the technique's branch and I'm closing this PR.

It just means that when (assuming) we get agreement on the technique, we merge both at the same time. (Last week we missed one and the example link went to a 404.)
Also, the link in the desc when to a whole other technique, and I hadn't realized!
